test/integration/react-streaming-and-server-components/test/index.test.js

```diff
@@ -295,15 +295,13 @@ async function runBasicTests(context, env) {
 
     const browser = await webdriver(context.appPort, '/next-api/link')
     await browser.eval('window.beforeNav = 1')
-    await browser.elementByCss('#next_id').click()
-    await browser.elementByCss('#next_id').click()
-    await check(() => browser.waitForElementByCss('#query').text(), /query:2/)
-
-    if (!isDev) {
-      // this might do a hard navigation in development
-      // so only require this check in production
-      expect(await browser.eval('window.beforeNav')).toBe(1)
-    }
+    await browser.waitForElementByCss('#next_id').click()
+    await waitFor(500)
```

Comment:
`waitFor` seems like reliable than `check` 🤔 should we do two separate checks the first checking for `query:1` and then another to check for `query:2`?